hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -301,13 +303,25 @@ public static String getUserAgent(Event event) {
   }
 
   public static Optional<String> getFullHttpUrl(Event event) {
-    if (event.getHttp() != null && event.getHttp().getRequest() != null) {
-      return Optional.ofNullable(event.getHttp().getRequest().getUrl());
+    if (event.getHttp() != null
+        && event.getHttp().getRequest() != null
+        && event.getHttp().getRequest().getUrl() != null
+        && isAbsoluteUrl(event.getHttp().getRequest().getUrl())) {
+      return Optional.of(event.getHttp().getRequest().getUrl());
     }
-
     return Optional.empty();
   }
 
+  private static boolean isAbsoluteUrl(String urlStr) {
+    try {
+      URL url = new URL(new URL("http://hypertrace.org"), urlStr);
```

Comment:
Why do we need this `http://hypertrace.org`?

Reply:
Any other url will be fine too. In case when input url is relative, this provided url makes up for the scheme, authority etc.